.golangci.yml

```diff
@@ -0,0 +1,91 @@
+linters-settings:
+#  govet:
+#    check-shadowing: true
+  golint:
+    min-confidence: 0.8
+  gocyclo:
+    min-complexity: 50
+  staticcheck:
+    go: 1.12
+#  maligned:
+#    suggest-new: true
+#  dupl:
+#    threshold: 150
+#  depguard:
+#    list-type: blacklist
+#    include-go-root: false
+#    packages:
+#      - github.com/sirupsen/logrus
+#      - github.com/bmizerany/assert
+#  funlen:
+#    lines: 70
+#    statements: 40
+#  lll:
+#    line-length: 150
+#  nakedret:
+#    max-func-lines: 10
+
+linters:
+  disable-all: true
+  enable:
+#    - asciicheck
+#    - bodyclose
+#    - deadcode
+#    - depguard
+#    - dogsled
+#    - dupl
+#    - errcheck
+#    - exhaustive
+#    - funlen
+#    - gochecknoglobals
+#    - gochecknoinits
+#    - goconst
+#    - gocritic
```

Comment:
> The most opinionated Go source code linter

In my opinion one of the best linters for Go. Finds a variety of things from "suspicious" `if` statements (potential bug), replacing `i = i + 1` with `i++`, using switch statements instead of long if .. else if .. else cases to missing spaces after `//` in comments. Different checks can be disabled per config.